build.gradle

```diff
@@ -1177,6 +1177,31 @@ project(':streams') {
     testRuntime libs.slf4jlog4j
   }
 
+  task processMessages(type:JavaExec) {
+    main = "org.apache.kafka.message.MessageGenerator"
+    classpath = project(':generator').sourceSets.main.runtimeClasspath
+    args = [ "org.apache.kafka.streams.internals.generated",
+             "src/generated/java/org/apache/kafka/streams/internals/generated",
+             "src/main/resources/common/message" ]
+    inputs.dir("src/main/resources/common/message")
+    outputs.dir("src/generated/java/org/apache/kafka/streams/internals/generated")
+  }
+
+  sourceSets {
+    main {
+      java {
+        srcDirs = ["src/generated/java", "src/main/java"]
+      }
+    }
+    test {
+      java {
+        srcDirs = ["src/generated/java", "src/test/java"]
+      }
+    }
+  }
+
+  compileJava.dependsOn 'processMessages'
```

Comment:
Unfortunately, IDEA doesn't pick this up, but you can run this manually before running in IDEA:
```
./gradlew :clients:processMessages :clients:processTestMessages :streams:processMessages
```

Reply:
yeah, that'll do it, too, as will `:compile`. The line above is just the minimal command to run (also works on the client code that needs to be re-generated every time you switch branches). It takes like 1.5 seconds.

Reply:
Yes, this line makes the code generation a dependency of the compile phase, which also has to run before the tests.